src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/BuildPropertyPage/VBWarningsValueProvider.cs

```diff
@@ -0,0 +1,292 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties;
+
+/// <remarks>
+/// <para>
+/// The VB "Compile" property page doesn't allow you to disable individual warnings
+/// or promote individual warnings to errors. Instead, it groups related warnings
+/// into sets and lets the user adjust the severity of the whole set at once. This
+/// type is responsible for mapping between those sets and the actual property
+/// values persisted to the project file.
+/// </para>
+/// <para>
+/// Note this types stores IDs in instances of <see cref="ImmutableSortedSet{T}"/>,
+/// rather a simple array or list. Two reasons:
+/// <list type="number">
+/// <item>We need to perform actual set operations (union, difference, etc.)</item>
+/// <item>When writing IDs to the project file, we want to keep them sorted</item>
+/// </list>
+/// </para>
+/// </remarks>
+[ExportInterceptingPropertyValueProvider(
+    new[]
+    {
+        ImplicitConversionPropertyName,
+        LateBindingPropertyName,
+        ImplicitTypePropertyName,
+        UseOfVariablePriorToAssignmentPropertyName,
+        ReturningRefTypeWithoutReturnValuePropertyName,
+        ReturningIntrinsicValueTypeWithoutReturnValuePropertyName,
+        UnusedLocalVariablePropertyName,
+        InstanceVariableAccessesSharedMemberPropertyName,
+        RecursiveOperatorOrPropertyAccessPropertyName,
+        DuplicateOrOverlappingCatchBlocksPropertyName
+    },
+    ExportInterceptingPropertyValueProviderFile.ProjectFile)]
+internal sealed class VBWarningsValueProvider : InterceptingPropertyValueProviderBase
+{
+    // The names of the property "sets" as known to the property page.
+    internal const string ImplicitConversionPropertyName = "ImplicitConversion";
```

Comment:
Just to confirm: none of these literal values can be sourced from generated rule file constants? Using such constants helps make explicit the otherwise implicit relationship between them, causing compile errors if we remove the properties from the rule, for example.

Reply:
I'll double check. Note that the system doesn't use `Rule`s at this level* (i.e. we don't need a `Rule` to define which properties we can access), so I don't want to introduce a false dependency, either.

*That's a lie, but true enough for the purposes of this discussion. :-)